hack/tpm2/main.go

```
@@ -0,0 +1,809 @@
+			ObjectAttributes: tpm2.TPMAObject{
+				Decrypt:      true,
+				SignEncrypt:  true,
+				UserWithAuth: true,
+			},
+			Parameters: tpm2.NewTPMUPublicParms(tpm2.TPMAlgRSA, &tpm2.TPMSRSAParms{
+				Symmetric: tpm2.TPMTSymDefObject{
+					Algorithm: tpm2.TPMAlgNull,
+					Mode:      tpm2.NewTPMUSymMode(tpm2.TPMAlgRSA, tpm2.TPMAlgNull),
+				},
+				Scheme: tpm2.TPMTRSAScheme{
+					Scheme: tpm2.TPMAlgNull,
+					Details: tpm2.NewTPMUAsymScheme(tpm2.TPMAlgRSA, &tpm2.TPMSSigSchemeRSASSA{
+						HashAlg: tpm2.TPMAlgNull,
+					}),
+				},
+				KeyBits:  tpm2.TPMKeyBits(pubKey.N.BitLen()),
+				Exponent: uint32(pubKey.E),
+			}),
+			Unique: tpm2.NewTPMUPublicID(tpm2.TPMAlgRSA, &tpm2.TPM2BPublicKeyRSA{
+				Buffer: pubKey.N.Bytes(),
+			}),
+		}),
+	}
+
+	loadExternalResponse, err := loadExternal.Execute(t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load external key: %v", err)
+	}
+
+	defer func() {
+		flush := tpm2.FlushContext{
+			FlushHandle: loadExternalResponse.ObjectHandle,
+		}
+
+		_, flushErr := flush.Execute(t)
+		if flushErr != nil {
+			err = flushErr
+		}
+	}()
+
+	pcrSelector, err := createPCRSelection([]int{11})
+	if err != nil {
+		return nil, err
+	}
+
+	policyPCR := tpm2.PolicyPCR{
+		PolicySession: policySess.Handle(),
+		Pcrs: tpm2.TPMLPCRSelection{
+			PCRSelections: []tpm2.TPMSPCRSelection{
+				{
+					Hash:      tpm2.TPMAlgSHA256,
+					PCRSelect: pcrSelector,
+				},
+			},
+		},
+	}
+
+	_, err = policyPCR.Execute(t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute policy pcr: %v", err)
+	}
+
+	policyGetDigest := tpm2.PolicyGetDigest{
+		PolicySession: policySess.Handle(),
+	}
+
+	policyGetDigestResp, err := policyGetDigest.Execute(t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute policy get digest: %v", err)
+	}
+
+	var sigJSON PCRData
+
+	if err := json.Unmarshal(pcrDataJSON, &sigJSON); err != nil {
+		return nil, err
+	}
+
+	// TODO: verify the fingerprint matches the one in the JSON
```

this is just a fail early situation